src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -469,11 +470,18 @@ public void download() {
                     linkedFile.setFileType(suggestedTypeName);
                     String suggestedName = linkedFileHandler.getSuggestedFileName(externalFileType.getExtension());
                     String fulltextDir = FileUtil.createDirNameFromPattern(databaseContext.getDatabase(), entry, filePreferences.getFileDirectoryPattern());
-                    suggestedName = FileNameUniqueness.getNonOverWritingFileName(targetDirectory.resolve(fulltextDir), suggestedName);
-                    return targetDirectory.resolve(fulltextDir).resolve(suggestedName);
+                    String finalName = FileNameUniqueness.getNonOverWritingFileName(targetDirectory.resolve(fulltextDir), suggestedName);
+                    if (!suggestedName.equals(finalName)) {
+                        messageError.append("File: \"");
```

Comment:
StringBuilder is not necessary, I would simply add a localized string .e.g
messageError = Localization.lang("File with the name %0 already exists. Aborting download.", suggestedName)

Reply:
I first used a string but I need to modify the error message depending on the error and I am no able to modify and string if I use it in a lambda because it needs to be final or at least have the behavior of a final variable if it is used at a lambda expression. That's why I decided to use a StringBuilder

Reply:
I leave it like this?

Reply:
Seems like the StringBuilder is the only option. However,  please use a localized error description, you might want to check the l10n/en.properties to check if there is already a message you can reuse

Reply:
Done, now is using the `Localization.lang()` for the error message